src/tileSplitterManager.ts

```diff
@@ -1,46 +1,58 @@
+import { join } from 'path';
 import { Logger } from '@map-colonies/js-logger';
 import { inject, singleton } from 'tsyringe';
 import { QueueClient } from './clients/queueClient';
 import { SERVICES } from './common/constants';
-import { IConfig } from './common/interfaces';
-
-// interface IParameters {
-//   resourceId: string;
-//   resourceVersion: string;
-//   layerRelativePath: string;
-// }
+import { IConfig, IGenerateTilesConfig, IS3Config, ITaskParams, ITilesConfig, IVrtConfig } from './common/interfaces';
+import { GDALUtilities } from './gdalUtilities';
+import { StorageProviderType } from './common/enums';
 
 @singleton()
 export class TileSplitterManager {
   private readonly splitAttempts: number;
+  private readonly storageProvider: string;
 
   public constructor(
     @inject(SERVICES.LOGGER) private readonly logger: Logger,
     @inject(SERVICES.CONFIG) private readonly config: IConfig,
-    // @inject(SERVICES.TILES_CONFIG) private readonly tilesConfig: ITilesConfig,
-    // @inject(SERVICES.STORAGE_PROVIDER) private readonly storageProvider: IStorageProvider,
+    @inject(SERVICES.S3_CONFIG) private readonly s3Config: IS3Config,
+    @inject(SERVICES.VRT_CONFIG) private readonly vrtConfig: IVrtConfig,
+    @inject(SERVICES.GENERATE_TILES_CONFIG) private readonly generateTilesConfig: IGenerateTilesConfig,
+    @inject(SERVICES.TILES_CONFIG) private readonly tilesConfig: ITilesConfig,
     private readonly queueClient: QueueClient
   ) {
     this.splitAttempts = this.config.get<number>('splitAttempts');
+    this.storageProvider = this.config.get<string>('storageProvider');
   }
 
   public async handleSplitTilesTask(): Promise<boolean> {
     const tilesTask = await this.queueClient.queueHandlerForTileSplittingTasks.dequeue();
     if (tilesTask) {
+      const gdalUtilities = new GDALUtilities(this.logger, this.vrtConfig, this.generateTilesConfig, this.queueClient, tilesTask);
       const jobId = tilesTask.jobId;
       const taskId = tilesTask.id;
       const attempts = tilesTask.attempts;
-      // const params = tilesTask.parameters as IParameters;
-      // const layerRelativePath = params.layerRelativePath;
+      const discreteId = (tilesTask.parameters as ITaskParams).discreteId;
```

Comment:
pass as parameter into GDALUtilities

Reply:
i've passed entire task to GDAL Utilities